ports/luasec/CMakeLists.txt

```
@@ -0,0 +1,41 @@
+cmake_minimum_required(VERSION 3.20.0)
+project(luasec)
+
+find_path(LUA_INCLUDE_DIR lua.h PATH_SUFFIXES lua)
+find_library(LUA_LIBRARY lua)
+find_package(OpenSSL)
+
+set(LUASEC_INCLUDES ${LUA_INCLUDE_DIR} src)
+set(LUASEC_LIBRARIES
+    ${LUA_LIBRARY}
+    OpenSSL::SSL
+    OpenSSL::Crypto
+    OpenSSL::applink
+    ws2_32)
+
+add_library(ssl
```

This library name looks like it would conflict with other ssl libraries (`libssl.so`/`ssl.dll`). What's the official upstream name?

```suggestion
add_library(luasec
```

---

Upstream expects the ssl.dll/ssl.so name.

If the code is linked to the C/C++ project that should be alright to change it, however it's not uncommon to include through lua in which case the ssl.dll name needs to stay.

https://packages.debian.org/bullseye/amd64/lua-sec/filelist is how debian packages it:
liblua<lua_version>-sec.so.<luasec_version> for linking to C and lua/<lua_version>/ssl.so for including in lua.

Any preferences? I think lpeg, luasocket, luafilesystem, etc. should follow the same convention.

---

archlinux only installs it to `usr/lib/lua/5.4/ssl.so` - any opinion on bin/lua/ssl.(dll|so)

---

https://github.com/lua/lua/blob/681297187ec45268e872b26753c441586c12bdd8/luaconf.h#L212 may also be relevant to the discussion if we're concerned about the dynamic load from lua use case. We could recommend setting CPATH to include lua-?.dll and name the library lua-ssl.dll

Do you like that option?

---

I'd like to change this name to `lua-ssl`.

---

Do you think we want or need any message letting people know to include lua-?.dll in their LUA_CPATH? I'm not sure the policy on messages in vcpkg.

---

- If this library completely uses the code of other third-party libraries, we must instead find and use the third-party libraries in vcpkg.
- If this library only conflicts with other libraries in name, I think in order to prevent the library name from conflicting with other libraries, we have to change its name to `${PORT}-${LIBRARY}` to avoid that.

---

ok, I have changed the library to lua-ssl.